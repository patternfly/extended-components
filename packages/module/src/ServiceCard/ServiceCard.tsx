import React from 'react';
import { Card, CardBody, CardFooter, CardHeader, Content, ContentVariants } from '@patternfly/react-core';
import { HelperText } from '@patternfly/react-core/dist/dynamic/components/HelperText';
import { HelperTextItem } from '@patternfly/react-core/dist/dynamic/components/HelperText';
import { createUseStyles } from 'react-jss';

export interface ServiceCardProps {
  /** Service card title */
  title: string;
  /** Service card subtitle */
  subtitle: string;
  /** Service card description */
  description: string;
  /** Service card icon */
  icon: React.ReactNode;
  /** Optional Service card helper text*/
  helperText?: string;
  /** Optional footer */
  footer?: React.ReactElement | null;
  /** Optional custom OUIA ID */
  ouiaId?: string | number;
}

const useStyles = createUseStyles({
  card: {
    height: 'var(--pf-v6-u-h-100)'
  },
  image: {
    marginRight: 'var(--pf-t--global--spacer--md)',
    width: 48
  }
});

const ServiceCard: React.FunctionComponent<ServiceCardProps> = ({
  title,
  subtitle,
  description,
  icon,
  helperText,
  footer = null,
  ouiaId='ServiceCard'
}: ServiceCardProps) => {
  const classes = useStyles();

  return (
    <Card className={classes.card} ouiaId={`${ouiaId}-card`}>
      <CardHeader>
        <div className={classes.image}>
          {icon}
        </div>
        <Content>
          <Content component={ContentVariants.h2} ouiaId={`${ouiaId}-title`}>{title}</Content>
          {subtitle}
        </Content>
      </CardHeader>
      <CardBody data-ouia-component-id={`${ouiaId}-description`}>{description}</CardBody>
      <CardFooter data-ouia-component-id={`${ouiaId}-footer`}>
        { helperText ?
          ( <HelperText data-ouia-component-id={`${ouiaId}-helper-text`}>
            <HelperTextItem className="pf-v6-u-mb-lg">
              {helperText}
            </HelperTextItem>
          </HelperText>) : null
        }
        { footer }
      </CardFooter>
    </Card>
  )
}

export default ServiceCard;