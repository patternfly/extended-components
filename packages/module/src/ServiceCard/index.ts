export { default } from './ServiceCard'
export * from './ServiceCard'
