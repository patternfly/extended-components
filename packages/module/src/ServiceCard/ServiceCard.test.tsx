import React from 'react';
import { render } from '@testing-library/react';
import ServiceCard from './ServiceCard';

describe('LogSnippet component', () => {
  it('should render LogSnippet component', () => {
    expect(render(<ServiceCard 
      title='Example'
      subtitle='A basic example'
      description='This is a basic ServiceCard Example'
      icon='/'
      helperText=''
    />)).toMatchSnapshot();
  });
});