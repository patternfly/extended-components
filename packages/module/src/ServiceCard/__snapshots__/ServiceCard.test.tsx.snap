// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard component should render ServiceCard component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v6-c-card"
        data-ouia-component-id="ServiceCard-card"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-l-flex pf-m-row pf-m-align-items-center"
            >
              <div
                class="image-0-2-2"
              >
                /
              </div>
              <div
                class=""
              >
                <div
                  class="pf-v6-c-content"
                  data-ouia-component-id="OUIA-Generated-Content-1"
                  data-ouia-component-type="PF6/Content"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  <h2
                    class="pf-v6-c-content--h2"
                    data-ouia-component-id="ServiceCard-title"
                    data-ouia-component-type="PF6/Content"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    Example
                  </h2>
                  A basic example
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-v6-c-card__body"
          data-ouia-component-id="ServiceCard-description"
        >
          This is a basic ServiceCard Example
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v6-c-card"
      data-ouia-component-id="ServiceCard-card"
      data-ouia-component-type="PF6/Card"
      data-ouia-safe="true"
      id=""
    >
      <div
        class="pf-v6-c-card__header"
      >
        <div
          class="pf-v6-c-card__header-main"
        >
          <div
            class="pf-v6-l-flex pf-m-row pf-m-align-items-center"
          >
            <div
              class="image-0-2-2"
            >
              /
            </div>
            <div
              class=""
            >
              <div
                class="pf-v6-c-content"
                data-ouia-component-id="OUIA-Generated-Content-1"
                data-ouia-component-type="PF6/Content"
                data-ouia-safe="true"
                data-pf-content="true"
              >
                <h2
                  class="pf-v6-c-content--h2"
                  data-ouia-component-id="ServiceCard-title"
                  data-ouia-component-type="PF6/Content"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  Example
                </h2>
                A basic example
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pf-v6-c-card__body"
        data-ouia-component-id="ServiceCard-description"
      >
        This is a basic ServiceCard Example
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
